src/k8s/pkg/k8sd/types/worker.go

```diff
@@ -14,6 +14,8 @@ type InternalWorkerNodeToken struct {
 	Secret string `json:"secret"`
 	// JoinAddresses is a list of control-plane addresses that exist in the cluster.
 	JoinAddresses []string `json:"join_addresses"`
+	// Fingerprint is used for server Auth
+	Fingerprint string `json:"fingerprint"`
```

Comment:
NOTE: we also need to set the fingerprint when generating the worker node join token